src/compiler/parser.ts

```diff
@@ -3544,19 +3543,19 @@ module ts {
             return finishNode(node);
         }
 
-        function parseArrayLiteralElement(): Expression {
+        function parseArgumentOrArrayLiteralElement(): Expression {
             return token === SyntaxKind.DotDotDotToken ? parseSpreadElement() : parseAssignmentExpressionOrOmittedExpression();

```

Comment:
Is parseAssignmentExpressionOrOmittedExpression called from anywhere else? If not, it might make sense to just fold it into here.